lib/worker/fake-tty.js

```diff
@@ -2,19 +2,30 @@
 const tty = require('tty');
 const options = require('./options').get();
 
-if (options.tty) {
-	Object.assign(process.stdout, {isTTY: true}, options.tty);
+const fakeTTYs = [];
 
-	if (options.tty.colorDepth !== undefined) {
-		process.stdout.getColorDepth = () => options.tty.colorDepth;
+const isatty = tty.isatty;
+tty.isatty = fd => {
+	if (fakeTTYs.indexOf(fd) !== -1) {
+		return true;
 	}
+	return isatty(fd);
+};
 
-	const isatty = tty.isatty;
-	tty.isatty = function (fd) {
-		if (fd === 1 || fd === process.stdout) {
-			return true;
-		}
+const simulateTTY = (stream, options) => {
+	Object.assign(stream, {isTTY: true}, options);
 
-		return isatty(fd);
-	};
+	if (options.colorDepth !== undefined) {
+		stream.getColorDepth = () => options.colorDepth;
+	}
+};
+
+if (options.tty.stderr) {
+	simulateTTY(process.stderr, options.tty.stderr);
+	fakeTTYs.push(2, process.stderr);
```

Comment:
Why do you need to push `process.stderr` (and `stdout`, below)?

In `isatty`, `fd` is supposed to be a number according to the Node.js documentation, so pushing the numbers should be sufficient.

Reply:
That makes sense to me. I included the `stdout` and `stderr` objects here to maintain consistency with the original code which checks both the numeric file descriptor and the stream object. If the tests pass then I'm happy to drop the stream object comparison.